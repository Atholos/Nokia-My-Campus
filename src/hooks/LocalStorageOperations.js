// Functions used with Local storage
const LocalStorageOperations=()=> {

    const myStorage = window.localStorage;

    // Add a new data,key value
    const create = (data,key) =>{
        myStorage.setItem(key,data);
        console.log(`Item Saved: ${data}`)
    };

    // Get an item/data corresponding the key
    const read = (key) =>{
        let item = myStorage.getItem(key);
        return JSON.parse(item)
    };

    const del = (key) =>{
        myStorage.removeItem(key);
        console.log(`Item deleted for key: ${key}`)
    };

    const clear = () =>{
      myStorage.clear();
        console.log("LocalStorage Cleared")
    };

    return {
        create,
        read,
        del,
        clear
    }
};

export default LocalStorageOperations
