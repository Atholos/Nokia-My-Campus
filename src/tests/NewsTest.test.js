import React from 'react'
import { createMount,createShallow } from '@material-ui/core/test-utils';
import News from './../views/news';
import HighlightItem from './../fragments/NewsHighlight';

//Needed imports for react redux testing
import { Provider } from 'react-redux'
import NewsReducer from "../reducers/NewsReducer";
import { createStore, combineReducers } from 'redux';
import LocalStorageOperations from '../hooks/LocalStorageOperations';


describe('News', () => {
  const { create } = LocalStorageOperations()
  //let json = {username: 'username', token: 'token'};
  //let user = create(JSON.stringify(json), 'user')

  const highlight = {
    title: "Halloween party",
    higlight: true,
    description:
      "Nokia halloween party this saturday at dream cafe! Come join us for some eerie fun",
    timestamp: "October 20, 2020",
    imgUrl: require("../assets/default.jpg"),
    imgTitle: "Halloween Party",
    paragraphs: {
      1: "This is the world we live in",
      2: "Come join us in death",
      3: "Hehou hahau",
      4: `Lorem ipsum dolor sit amet, meis expetenda ei eam, has ne assum aeque adipiscing, no legere utroque qui. Vix et congue invidunt facilisi, audire sententiae ea sea. Nulla accusam elaboraret eos ne. Solet putant epicuri te duo, per no error vivendum contentiones, legere mnesarchum duo ne. At qui sint dignissim, eum oratio veniam an.

    Probo ancillae vituperatoribus in mea, his an dico enim bonorum, illum gubergren posidonium ex quo. Sit ea case minimum, mei malorum accusam expetendis ex. Ei has exerci nemore convenire, duo id eirmod nonumes. Eos et animal dolorum epicurei, te eam altera mediocrem, eu sea torquatos reformidans.
    
    Elitr iudicabit eloquentiam cu vim. Graecis mediocritatem his in, ne erant harum soleat cum. In natum propriae mel. Quod impetus.`,
      5: `Lorem ipsum dolor sit amet, meis expetenda ei eam, has ne assum aeque adipiscing, no legere utroque qui. Vix et congue invidunt facilisi, audire sententiae ea sea. Nulla accusam elaboraret eos ne. Solet putant epicuri te duo, per no error vivendum contentiones, legere mnesarchum duo ne. At qui sint dignissim, eum oratio veniam an.

    Probo ancillae vituperatoribus in mea, his an dico enim bonorum, illum gubergren posidonium ex quo. Sit ea case minimum, mei malorum accusam expetendis ex. Ei has exerci nemore convenire, duo id eirmod nonumes. Eos et animal dolorum epicurei, te eam altera mediocrem, eu sea torquatos reformidans.
    
    Elitr iudicabit eloquentiam cu vim. Graecis mediocritatem his in, ne erant harum soleat cum. In natum propriae mel. Quod impetus.`,
    },
    paragraphImg: {
      2: require("../assets/default.jpg"),
      5: require("../assets/default.jpg"),
    },
  };
  
  //!!!!!!!!!!! TEST STORE FOR NEWS WITH REDUX !!!!!!!!!!!!!!
  const store = createStore(
    combineReducers({
      NewsReducer
    })
  );

  let mount;
  let shallow;

  beforeAll(() => {
    mount = createMount();
    shallow = createShallow();
  });

  afterAll(() => {
    mount.cleanUp();
    shallow.cleanUp();
  });

  it('Mounting News', () => {
    const tree = mount(<News/>);
    expect(tree).toMatchSnapshot();
  });

  it('testing highlight snapshot', () => {
    const tree = shallow(<Provider store={store}><HighlightItem highlight={highlight} /></Provider>);
    expect(tree).toMatchSnapshot();
  });
  
});