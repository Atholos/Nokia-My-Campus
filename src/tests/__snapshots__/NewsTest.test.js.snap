// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`News Mounting News 1`] = `
<News>
  <AuthNews>
    <AuthLoading>
      <div
        style={
          Object {
            "alignItems": "center",
            "bottom": 0,
            "display": "flex",
            "height": "100%",
            "justifyContent": "center",
            "left": 0,
            "margin": "auto 0",
            "position": "absolute",
            "right": 0,
            "textAlign": "center",
            "top": 0,
          }
        }
      >
        <WithStyles(ForwardRef(CircularProgress))
          style={Object {}}
        >
          <ForwardRef(CircularProgress)
            classes={
              Object {
                "circle": "MuiCircularProgress-circle",
                "circleDeterminate": "MuiCircularProgress-circleDeterminate",
                "circleDisableShrink": "MuiCircularProgress-circleDisableShrink",
                "circleIndeterminate": "MuiCircularProgress-circleIndeterminate",
                "circleStatic": "MuiCircularProgress-circleStatic",
                "colorPrimary": "MuiCircularProgress-colorPrimary",
                "colorSecondary": "MuiCircularProgress-colorSecondary",
                "determinate": "MuiCircularProgress-determinate",
                "indeterminate": "MuiCircularProgress-indeterminate",
                "root": "MuiCircularProgress-root",
                "static": "MuiCircularProgress-static",
                "svg": "MuiCircularProgress-svg",
              }
            }
            style={Object {}}
          >
            <div
              className="MuiCircularProgress-root MuiCircularProgress-colorPrimary MuiCircularProgress-indeterminate"
              role="progressbar"
              style={
                Object {
                  "height": 40,
                  "width": 40,
                }
              }
            >
              <svg
                className="MuiCircularProgress-svg"
                viewBox="22 22 44 44"
              >
                <circle
                  className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
                  cx={44}
                  cy={44}
                  fill="none"
                  r={20.2}
                  strokeWidth={3.6}
                  style={Object {}}
                />
              </svg>
            </div>
          </ForwardRef(CircularProgress)>
        </WithStyles(ForwardRef(CircularProgress))>
      </div>
    </AuthLoading>
  </AuthNews>
</News>
`;

exports[`News testing highlight snapshot 1`] = `
<ContextProvider
  value={
    Object {
      "store": Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      },
      "subscription": Subscription {
        "handleChangeWrapper": [Function],
        "listeners": Object {
          "notify": [Function],
        },
        "onStateChange": [Function],
        "parentSub": undefined,
        "store": Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        },
        "unsubscribe": null,
      },
    }
  }
>
  <HighlightItem
    highlight={
      Object {
        "description": "Nokia halloween party this saturday at dream cafe! Come join us for some eerie fun",
        "higlight": true,
        "imgTitle": "Halloween Party",
        "imgUrl": "default.jpg",
        "paragraphImg": Object {
          "2": "default.jpg",
          "5": "default.jpg",
        },
        "paragraphs": Object {
          "1": "This is the world we live in",
          "2": "Come join us in death",
          "3": "Hehou hahau",
          "4": "Lorem ipsum dolor sit amet, meis expetenda ei eam, has ne assum aeque adipiscing, no legere utroque qui. Vix et congue invidunt facilisi, audire sententiae ea sea. Nulla accusam elaboraret eos ne. Solet putant epicuri te duo, per no error vivendum contentiones, legere mnesarchum duo ne. At qui sint dignissim, eum oratio veniam an.

    Probo ancillae vituperatoribus in mea, his an dico enim bonorum, illum gubergren posidonium ex quo. Sit ea case minimum, mei malorum accusam expetendis ex. Ei has exerci nemore convenire, duo id eirmod nonumes. Eos et animal dolorum epicurei, te eam altera mediocrem, eu sea torquatos reformidans.
    
    Elitr iudicabit eloquentiam cu vim. Graecis mediocritatem his in, ne erant harum soleat cum. In natum propriae mel. Quod impetus.",
          "5": "Lorem ipsum dolor sit amet, meis expetenda ei eam, has ne assum aeque adipiscing, no legere utroque qui. Vix et congue invidunt facilisi, audire sententiae ea sea. Nulla accusam elaboraret eos ne. Solet putant epicuri te duo, per no error vivendum contentiones, legere mnesarchum duo ne. At qui sint dignissim, eum oratio veniam an.

    Probo ancillae vituperatoribus in mea, his an dico enim bonorum, illum gubergren posidonium ex quo. Sit ea case minimum, mei malorum accusam expetendis ex. Ei has exerci nemore convenire, duo id eirmod nonumes. Eos et animal dolorum epicurei, te eam altera mediocrem, eu sea torquatos reformidans.
    
    Elitr iudicabit eloquentiam cu vim. Graecis mediocritatem his in, ne erant harum soleat cum. In natum propriae mel. Quod impetus.",
        },
        "timestamp": "October 20, 2020",
        "title": "Halloween party",
      }
    }
  />
</ContextProvider>
`;
